nick_work/fdaaa_port.py

```diff
@@ -0,0 +1,372 @@
+import os
+from bs4 import BeautifulSoup
+import xmltodict
+import json
+from datetime import date
+from datetime import datetime
+from datetime import timedelta
+from dateutil.relativedelta import relativedelta
+import pandas as pd
+import csv
+cs = 'clinical_study'
+
+projectid = "ebmdatalab"
+
+fda_reg_df = pd.read_gbq('''
+SELECT *
+FROM clinicaltrials.jan17_fda_regulation_snapshot 
+''', projectid, dialect = 'standard')
+
+headers = ['nct_id', 'act_flag', 'included_pact_flag', 'has_results', 'pending_results', 'pending_data',\
+           'has_certificate', 'results_due', 'start_date', 'available_completion_date',\
+           'used_primary_completion_date', 'defaulted_pcd_flag', 'defaulted_cd_flag', 'results_submitted_date',\
+           'last_updated_date', 'certificate_date', 'phase', 'enrollment', 'location', 'study_status', 'study_type',\
+           'primary_purpose', 'sponsor', 'sponsor_type', 'collaborators', 'exported', 'fda_reg_drug',\
+           'fda_reg_device', 'is_fda_regulated', 'url', 'title', 'official_title', 'brief_title',\
+           'discrep_date_status', 'late_cert', ' defaulted_date', 'condition', 'condition_mesh', 'intervention',\
+           'intervention_mesh', 'keywords']
+
+effective_date = date(2017, 1, 18)
+directory = 'NCTxxx' #this points to a folder on my working directory. Change as needed.
+
+with open('test_output.csv', 'w') as test_csv:
+    writer = csv.writer(test_csv)
+    writer.writerow(headers)
+
+    for file in os.listdir(directory):
+        if file.endswith('.xml'):
+            xml_bs4 = open(directory + '/' + file)
```

Comment:
Great tips, thanks, Implemented. My first run at this I had not yet discovered `os.path.join()` but have since (and changed a whole bunch of other scripts I have) and just hadn't gotten around to updating this one yet!